OpenSim/Common/CSVAdapter.h

```diff
@@ -0,0 +1,161 @@
+ *                                                                            *
+ * Copyright (c) 2005-2015 Stanford University and the Authors                *
+ * Author(s): Ajay Seth                                                       *
+ * Contributors: Michael Sherman, Chris Dembia                                *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+/** @file
+ * This file defines concrete CSVFileReader and CSVFileWriter classes, which
+ * implements the  OpenSim::DataAdpater interface for reading or writing 
+ * CSV data files.
+*/
+#include "FileAdapter.h"
+
+#include <fstream>
+
+namespace OpenSim {
+
+//=============================================================================
+//=============================================================================
+/** CSVAdapter reads a Comma Separated Values file (.csv). 
+    First uncommented line is expected to be the column labels, also separated 
+    by commas. After that each row has the same number of numerical entries,
+    separated by commas. No blank lines or extra spaces are allowed. Here is an
+    example:
+<pre>
+time,x,y,z,theta,phi,psi
+0,1.2,-3,2.3,.02,1.03,.9
+.1,1.24,-2.9,2.8,.4,.99,.8
+1,.9,-2.3,2.2,.7,.08,.77
+</pre>
+**/
+
+
+template<typename DataType = SimTK::Real>
+class CSVAdapter_ : public FileAdapter {
+public:
+    using Table = TimeSeriesTable_<DataType>;
+
+    CSVAdapter_* clone() const override { 
+        return new CSVAdapter_(*this); 
+    }
+
+    void prepareForReading(AbstractDataTable& table) override {
+        table_ = &dynamic_cast<Table&>(table);
+    }
+
+    void prepareForWriting(const AbstractDataTable& table) override {
+        table_ = &dynamic_cast<Table&>(const_cast<AbstractDataTable&>(table));
+    }
+
+    void read() override;
+    
+    void write() override;
+
+    static std::string getIdentifier() {
+        return "csv";
+    }
+
+private:
+    Table* table_;
+    
+    static const std::string delimiter_read_;
+    static const std::string delimiter_write_;
+    static const std::string time_column_label_;
```

Comment:
Naming convention for class members (data and methods) is `camelCaseWithInitialLowerCase`; we use underscore separation only in property names and when we're trying to match STL, like in a container that has a `size_type` member. (This is separate from the data member prefix discussed in #533.) So conforming names for these could be:
```c++
static const std::string m_delimiterRead;
static const std::string m_delimiterWrite;
static const std::string m_timeColumnLabel;
```